### ISSUE TITLE
How do we Bring Back Google Drive As Kodi Source

### ISSUE THREAD

**Issue by 4ld3v** [cguZZman/plugin.googledrive#119]
**Is your feature request related to a problem? Please describe.**
@cguZZman This is about the google cloud addon source mode, that seems to be gone now due to the API changes made by google. Can you elaborate in the technical issue behind this is? Is this gone for good, or are there possibilities we havent discovered yet to bring this back.

Regards and Thanks for this awesome addon!

**Comment by cguZZman** [comment]
_From previous response:_
I found a possible solution to get it working again using the transcode url, but subtitles won't work. They may need to be downloaded or something to get it working. Still working on that part.
Anyway, there is hope for it. No ETA tho.
Just out of curiosity, why do you prefer source mode over .strm files?

**Comment by cguZZman** [comment]
The technical reason is that Google removed the option to send the access token authentication directly in the url of the request of the Drive API resource.
So when Kodi requested the download of a file, the addon was redirecting it to the google server, passing the access token in the url. Now Google requires the access token to be pass as a header, but a redirect response does not support a redirect of the authentication header.

**Comment by cguZZman** [comment]
I will keep this issue open for any future update

**Comment by 4ld3v** [comment]
tbh, i didnt know that this is the way your implementation does work - and i really like the way you solved the gdrive as a source. It is a very intresting problem the API update brought up and I am very curious to see then what you will come up with to fix that :)

As far as it concerns the  .strm files, I wasnt aware of this and saw this mentioned in another thread today. Thanks for reminding me and I will look into this.

**Comment by xybydy** [comment]
@cguZZman can you make a branch with the progress you've made? maybe I could help to figure the problem out

**Comment by IshanArya** [comment]
> Just out of curiosity, why do you prefer source mode over .strm files?

What's the difference between source mode and .strm files?

**Comment by xybydy** [comment]
AFAIK, Source is dead. .strm are locally created files linked to the remote medias.

**Comment by IshanArya** [comment]
So source is the processed video files that Google generates for web viewing?

**Comment by xybydy** [comment]
No, it is not. That's strm files method. Source is whole different thing.

**Comment by IshanArya** [comment]
What is source then?
